trace/google/cloud/trace/trace_span.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""TraceSpan for sending traces to the Stackdriver Trace API."""
+
+from google.cloud.gapic.trace.v1.enums import TraceSpan as Enum
+from google.protobuf.timestamp_pb2 import Timestamp
+
+import random
+
+
+class TraceSpan(object):
+    """A span is an individual timed event which forms a node of the trace tree. 
+    Each span has its name, span id and parent id. The parent id indicates the 
+    causal relationships between the individual spans in a single distributed trace.
+    Span that does not have a parent id is called root span. All spans associated
+    with a specific trace also share a common trace id. Spans do not need to be 
+    continuous, there can be gaps between two spans.
+
+    See
+    https://cloud.google.com/trace/docs/reference/v1/rpc/google.devtools.cloudtrace.v1
+    #google.devtools.cloudtrace.v1.TraceSpan
+
+    :type name: str
+    :param name: The name of the span.
+
+    :type kind: :class:`~google.cloud.trace.span.SpanKind`
+    :param kind: Distinguishes between spans generated in a particular context.
+                 For example, two spans with the same name may be distinguished using
+                 RPC_CLIENT and RPC_SERVER to identify queueing latency associated
+                 with the span.
+
+    :type parent_span_id: str
+    :param parent_span_id: ID of the parent span. Optional.
+
+    :type labels: dict
+    :param labels: Collection of labels associated with the span.
+                   Label keys must be less than 128 bytes.
+                   Label values must be less than 16 kilobytes.
+
+    :type span_id: str
+    :param span_id: Identifier for the span, unique within a trace.
+    """
+
+    def __init__(
+            self,
+            name,
+            kind=Enum.SpanKind.SPAN_KIND_UNSPECIFIED,
+            parent_span_id=None,
```

Comment:
It seems to me that if spans can have children, it should be possible to start a span from a span, e.g.:

```python

with trace_client.trace() as trace:
      with trace.span('root_span_1') as span:
            do_something()
            with span.span('child_span'):
                do_something()
      with trace.span('root_span_2'):
            do_something()
```